src/UNL/MediaHub/Media.php

```diff
@@ -442,10 +473,122 @@ public function userHasPermission(UNL_MediaHub_User $user, $permission_id)
         return false;
     }
     
-    public function getTextTracks()
+    public function getTextTrackURLs()
+    {
+        $track = UNL_MediaHub_MediaTextTrack::getById($this->media_text_tracks_id);
+        
+        if (!$track) {
+            return array();
+        }
+        
+        $files = $track->getFiles();
+        
+        if (empty($files->items)) {
+            return array();
+        }
+        
+        
+        $urls = array();
+        foreach ($files->items as $file) {
+            $urls[$file->language] = $file->getURL();
+        }
+        
+        return $urls;
+    }
+    
+    public function getAmaraTextTracks($format = 'vtt')
     {
         $api = new UNL_MediaHub_AmaraAPI();
-        return $api->getMediaHubTextTracks($this->id, $this->url);
+
+        return $api->getTextTracks($this->url, $format);
+    }
+    
+    public function findDuration()
+    {
+        if (!$this->getLocalFileName()) {
+            //We need the media to be local to find the duration
+            return false;
+        }
+        
+        $getId3 = new \GetId3\GetId3Core();
+        $details = $getId3->analyze($this->getLocalFileName());
+
+        if (!isset($details['playtime_string'])) {
+            return false;
+        }
+        
+        if (empty($details['playtime_string'])) {
+            return false;
+        }
+
+        //Convert to a standard string
+        switch (substr_count($details['playtime_string'], ':')) {
+            case 0:
+                $playtime_string = '00:00:' . str_pad($details['playtime_string'], 2, '0', STR_PAD_LEFT);
+                break;
+            case 1:
+                $playtime_string = '00:' . str_pad($details['playtime_string'], 5, '0', STR_PAD_LEFT);
+                break;
+            default:
+                $playtime_string = str_pad($details['playtime_string'], 8, '0', STR_PAD_LEFT);
+        }
+        
+        return array(
+            'string' => $playtime_string,
+            'seconds' => strtotime($playtime_string) - strtotime('TODAY')
```

Comment:
Ah, I remember why I'm not using `time()` here.

The idea is this:
By calling `strtotime()` on the playtime string, you get the unix time code for today + the length of the video. We needs the number of sends tho, so I can simply subtract the number of seconds since the START of the day. `time()` returns the CURRENT time.